src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1213,10 +1213,18 @@ def _need_update_container(self):
 
     def set_up_existing_container_update(self):
         if self.get_argument_force_single_container_updates():
+            if len(self.new_containerapp["properties"]["template"]["containers"]) == 0:
+                raise ValidationError(
+                    "The container app must have at least one container. Please use --image to set the image for the container app.")
```

Comment:
@snehapar9  @daniv-msft 

I don't understand why need to add this validation logic here. What is the background?
 What if the containerapp has no container by some mistake but need to fix it with command?

Reply:
Synced offline and provided more context.